functions/functionsv2/helloauditlog/hello_auditlog_test.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package helloworld provides a set of Cloud Functions samples.
+package helloworld
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/cloudevents/sdk-go/v2/event"
+	auditevents "github.com/googleapis/google-cloudevents-go/cloud/audit/v1"
+	"google.golang.org/protobuf/proto"
+)
+
+func makeAuditLog(subject string, payload auditevents.ProtoPayload) (event.Event, error) {
+	logevent := auditevents.LogEntryData{
+		ProtoPayload: &payload,
+	}
+	e := event.New()
+	e.SetSubject(subject)
+	e.SetType("google.cloud.audit.log.v1.written")
+	eventdata, err := json.Marshal(logevent)
+	if err != nil {
+		return event.New(), err
+	}
+	e.SetDataContentType("application/json")
+	e.SetData("application/json", eventdata)
+	return e, nil
+}
+
+func TestHelloAuditLog(t *testing.T) {
+
+	tests := []struct {
+		name         string
+		subject      string
+		payload      auditevents.ProtoPayload
+		expectedLogs []string
+	}{
+		{"sample-output",
```

Comment:
suggestion: name declarations for longer struct definitions improve readability. I went looking for whether we have a standard on this, and I couldn't find one so it may just be opinion ;)

Since I'm curious if I'm missing something, @codyoss do you know?
I skimmed through https://golang.org/doc/effective_go and https://github.com/golang/go/wiki/CodeReviewComments

Reply:
convention or not, i'm happy to add that. it seems like a reasonable practice, even if it is not standard. (will leave comment open for cody's input).